addons/mail/wizard/mail_compose_message.py

```diff
@@ -819,3 +851,45 @@ def _generate_template_for_composer(self, template, res_ids, render_fields,
             }
 
         return render_results
+
+    # ----------------------------------------------------------------------
+    # MISC UTILS
+    # ----------------------------------------------------------------------
+
+    @api.model
+    def _parse_res_domain(self, domain_str):
+        """ Parse domain for composer, which can be: an already valid list or
+        tuple (generally in code), a list or tuple as a string (coming from
+        actions). Void strings are considered as a falsy domain. """
+        if isinstance(domain_str, (list, tuple)):
+            return domain_str
+        if not domain_str:
+            return expression.FALSE_DOMAIN
+        error_msg = _("Invalid domain %(domain_str)r (type %(domain_type)s)",
+                      domain_str=domain_str,
+                      domain_type=type(domain_str))
+        try:
+            domain = ast.literal_eval(domain_str)
+        except Exception as e:
+            raise ValidationError(error_msg) from e
+        if not isinstance(domain, (list, tuple)):
+            raise ValidationError(error_msg)
+        return domain
+
+    @api.model
+    def _parse_res_ids(self, res_ids_str):
+        """ Parse res_ids for composer, which can be: an already valid list or
+        tuple (generally in code), a list or tuple as a string (coming from
+        actions). Void strings are returned as it, to be managed by the caller. """
+        if isinstance(res_ids_str, (list, tuple)) or not res_ids_str:
+            return res_ids_str
+        error_msg = _("Invalid res_ids %(res_ids_str)s (type %(res_ids_type)s)",
+                      res_ids_str=res_ids_str,
+                      res_ids_type=type(res_ids_str))
+        try:
+            res_ids = ast.literal_eval(res_ids_str)
+        except Exception as e:
+            raise ValidationError(error_msg) from e
+        if not isinstance(res_ids, (list, tuple)):
+            raise ValidationError(error_msg)
```

Comment:
someone added a new function in `tools/misc.py` that can be used here to even check that's a list of integer :disguised_face: 
```python
from odoo.tools import is_list_of

if not is_list_of(res_ids, int):
   ...
```

Reply:
Someone smart ?